drivers/wasapi/audio_driver_wasapi.cpp

```diff
@@ -453,12 +440,18 @@ Error AudioDriverWASAPI::audio_device_init(AudioDeviceWASAPI *p_device, bool p_c
 		buffer_frames = period_frames;
 
 		hr = device_audio_client_3->InitializeSharedAudioStream(0, period_frames, pwfex, nullptr);
-		ERR_FAIL_COND_V_MSG(hr != S_OK, ERR_CANT_OPEN, "WASAPI: InitializeSharedAudioStream failed with error 0x" + String::num_uint64(hr, 16) + ".");
-		uint32_t output_latency_in_frames;
-		WAVEFORMATEX *current_pwfex;
-		device_audio_client_3->GetCurrentSharedModeEnginePeriod(&current_pwfex, &output_latency_in_frames);
-		real_latency = (float)output_latency_in_frames / (float)current_pwfex->nSamplesPerSec;
-		CoTaskMemFree(current_pwfex);
+		if (hr != S_OK) {
+			print_verbose("WASAPI: InitializeSharedAudioStream failed with error 0x" + String::num_uint64(hr, 16) + ", falling back to IAudioClient.");
+			CoTaskMemFree(pwfex);
+			SAFE_RELEASE(device);
+			return audio_device_init(p_device, p_capture, reinit, true);
+		} else {
+			uint32_t output_latency_in_frames;
+			WAVEFORMATEX *current_pwfex;
+			device_audio_client_3->GetCurrentSharedModeEnginePeriod(&current_pwfex, &output_latency_in_frames);
```

Comment:
Probably should have the same check for the `GetCurrentSharedModeEnginePeriod`, just in case some future version of wine will IAudioClent3 partially implemented.